Automattic/portkey-android
# Introduce mp4composer lib as a module

NOTE: Please Review and Merge #23, #22 and #21 first  - do not merge until #21 gets merged here

Addresses #10 

- Started by making sure permissions for recording (writing to a file, capturing camera and audio) are set. (52f8c96, aa58b09)
-  introduced the library as a module in c43a742
- re-introduced `GlGifWatermarkFilter` in 156a596
- added video saving methods back in df05539

To test:
0. open the app then tap on the floating buton to go to the composing screen
1. on the overflow menu, tap on Camera preview once, allow all permissions
2. tap on Camera preview again to start recorrding
3. tap once more to stop recording
4. add emojis, text, drawing
5. on the overflow menu, tap Save
6. see the new video should have been saved to the device

ToDo:
- [x] fix style violations - code converted to Kotlin & fixed corresponding style violations in #21
